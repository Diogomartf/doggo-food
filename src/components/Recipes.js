import React from 'react';
import styled from 'styled-components';
import Recipe from './Recipe';

const recipes = {
  title: 'Recipe Puppy',
  version: 0.1,
  href: 'http://www.recipepuppy.com/',
  results: [
    {
      title: 'Basil Pesto Pomodoro Sauce',
      href: 'http://www.recipezaar.com/Basil-Pesto-Pomodoro-Sauce-221792',
      ingredients: 'pesto, nonstick cooking spray, tomato, onions',
      thumbnail: 'http://img.recipepuppy.com/148477.jpg'
    },
    {
      title: "Ali's Amazing Bruschetta",
      href: 'http://allrecipes.com/Recipe/Alis-Amazing-Bruschetta/Detail.aspx',
      ingredients: 'olive oil, parmesan cheese, onions, tomato',
      thumbnail: 'http://img.recipepuppy.com/30661.jpg'
    },
    {
      title: 'BLT Pasta Salad',
      href: 'http://allrecipes.com/Recipe/BLT-Pasta-Salad/Detail.aspx',
      ingredients: 'bacon, ranch dressing, onions, tomato',
      thumbnail: 'http://img.recipepuppy.com/7095.jpg'
    },
    {
      title: 'Sauce Dijon a la Provencal \r\n\t\t\n',
      href: 'http://www.kraftfoods.com/kf/recipes/sauce-dijon-a-la-54823.aspx',
      ingredients: 'tomato, onions, dijon mustard, sugar',
      thumbnail: 'http://img.recipepuppy.com/601515.jpg'
    },
    {
      title: 'Tomato Alfredo Sauce with Artichokes',
      href:
        'http://allrecipes.com/Recipe/Tomato-Alfredo-Sauce-with-Artichokes/Detail.aspx',
      ingredients: 'flour, whole milk, onions, tomato',
      thumbnail: 'http://img.recipepuppy.com/11220.jpg'
    },
    {
      title: 'Tomato Cheese Crunch',
      href: 'http://allrecipes.com/Recipe/Tomato-Cheese-Crunch/Detail.aspx',
      ingredients: 'cheddar cheese, onions, potato chips, tomato',
      thumbnail: 'http://img.recipepuppy.com/20868.jpg'
    },
    {
      title: 'Basic Guacamole Dip',
      href: 'http://allrecipes.com/Recipe/Basic-Guacamole-Dip/Detail.aspx',
      ingredients: 'avocado, lemon juice, onions, tomato',
      thumbnail: 'http://img.recipepuppy.com/29413.jpg'
    }
  ]
};

const Recipes = ({ searchText }) => (
  <Container>
    <Title>Recipes with {searchText}</Title>
    <Grid>
      {recipes.results.map(recipe => (
        <Recipe
          key={recipe.href}
          title={recipe.title}
          ingredients={recipe.ingredients}
          url={recipe.href}
          image={recipe.thumbnail}
        />
      ))}
    </Grid>
  </Container>
);

const Grid = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-column-gap: 24px;
  grid-row-gap: 24px;

  @media (min-width: 768px) {
    grid-template-columns: 1fr 1fr;
  }

  @media (min-width: 998px) {
    grid-template-columns: 1fr 1fr 1fr;
  }
`;

const Container = styled.div`
  padding: 12px 24px;
`;

const Title = styled.div`
  margin-bottom: 16px;

  font-size: 22px;
  font-weight: 600;
  line-height: 26px;
  color: ${({ theme }) => theme.colors.black};
`;

export default Recipes;
