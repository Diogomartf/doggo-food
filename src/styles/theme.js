import colors from './colors';

export default {
  name: 'default theme',
  colors
};
