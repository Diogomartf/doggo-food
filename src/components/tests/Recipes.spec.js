import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../../styles';

import Recipes from '../Recipes';

const recipes = [
  {
    title: 'Alfredo Recipe',
    href:
      'http://www.grouprecipes.com/77213/alfredo-sauce-weight-watchers.html',
    ingredients: 'garlic, pesto, water',
    thumbnail: 'http://img.recipepuppy.com/306873.jpg'
  },
  {
    title: "Grandma Sugar's Shell Macaroni and Cheese",
    href:
      'http://www.recipezaar.com/Grandma-Sugars-Shell-Macaroni-and-Cheese-311887',
    ingredients:
      'black pepper, butter, pasta (in general), milk, cheddar cheese, velveeta cheese',
    thumbnail: 'http://img.recipepuppy.com/683295.jpg'
  },
  {
    title: 'Cheese-Stuffed Shells in Marinara Sauce',
    href:
      'http://allrecipes.com/Recipe/Cheese-Stuffed-Shells-in-Marinara-Sauce-2/Detail.aspx',
    ingredients:
      'ricotta cheese, black pepper, parmesan cheese, pasta (in general), eggs, mozzarella cheese',
    thumbnail: 'http://img.recipepuppy.com/1947.jpg'
  }
];

describe('<Recipe/>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Recipes recipes={recipes} searchText={'cheese'} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render EmptyState when recipes length is 0', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Recipes recipes={[]} searchText={'grass'} />
      </ThemeProvider>
    );

    expect(wrapper.find('EmptyState')).toHaveLength(1);
  });

  it('should render 3 recipes', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Recipes recipes={recipes} searchText={'cheese'} />
      </ThemeProvider>
    );

    expect(wrapper.find('Recipe')).toHaveLength(3);
  });
});
