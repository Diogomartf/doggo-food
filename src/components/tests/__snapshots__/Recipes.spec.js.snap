// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipe/> should render correctly 1`] = `
<styled.div>
  <styled.div>
    Recipes with 
    cheese
  </styled.div>
  <styled.div>
    <Recipe
      image="http://img.recipepuppy.com/306873.jpg"
      ingredients="garlic, pesto, water"
      key="http://www.grouprecipes.com/77213/alfredo-sauce-weight-watchers.html"
      title="Alfredo Recipe"
      url="http://www.grouprecipes.com/77213/alfredo-sauce-weight-watchers.html"
    />
    <Recipe
      image="http://img.recipepuppy.com/683295.jpg"
      ingredients="black pepper, butter, pasta (in general), milk, cheddar cheese, velveeta cheese"
      key="http://www.recipezaar.com/Grandma-Sugars-Shell-Macaroni-and-Cheese-311887"
      title="Grandma Sugar's Shell Macaroni and Cheese"
      url="http://www.recipezaar.com/Grandma-Sugars-Shell-Macaroni-and-Cheese-311887"
    />
    <Recipe
      image="http://img.recipepuppy.com/1947.jpg"
      ingredients="ricotta cheese, black pepper, parmesan cheese, pasta (in general), eggs, mozzarella cheese"
      key="http://allrecipes.com/Recipe/Cheese-Stuffed-Shells-in-Marinara-Sauce-2/Detail.aspx"
      title="Cheese-Stuffed Shells in Marinara Sauce"
      url="http://allrecipes.com/Recipe/Cheese-Stuffed-Shells-in-Marinara-Sauce-2/Detail.aspx"
    />
  </styled.div>
</styled.div>
`;
